pkgs/build-support/build-incremental.nix

```diff
@@ -0,0 +1,54 @@
+{ pkgs }:
+rec {
+  /* Prepare a derivation for local builds.
+    *
+    * This function prepares incremental builds by provinding,
+    * containing the build output and the sources for cross checking.
+    * The build output can be used later to allow incremental builds
+    * by passing the derivation output to the `mkIncrementalBuild` function.
+    *
+    * To build a project incrementaly follow these steps:
```

Comment:
Maybe this more concrete example should go into the manual?

Reply:
I have integrated it into the manual.